File: TradebullsAngular/src/app/Models/Sales/ProcessInModel.ts
export class ProcessIn_AddModel {
    ProcessInID?:number;
    ProcessInNo?:string;
    ProcessInDate?:Date;
    ProcessInType?:string;
    ProcessID?:number;
    PartyID?:number;
    Remarks?:string;
    TotalQty?:number;
    Items:ProcessIn_Item_AddModel[] = [];
}

export class ProcessIn_Item_AddModel {
    ID?:number;
    WOPlanningItemComponentID: number = 0;
    ProcessOutItemID: number = 0;
    WONo: string = "";
    ProcessOutNo: string = "";
    SrNo?:number;
    ParentProductID?:number;
    ProductID?:number;
    sku?: string;
    ProductName?: string;
    ComponentID?:number;
    ComponentName?: string;
    Qty?:number;
    DQty?:number;
    RQty?:number;
    Remarks?:string;
    Photo: string = "";
}

export class ProcessIn_ProcessOut_ViewModel {
    ProcessOutID?:number;
    ProcessOutNo?:string;
    ProcessOutDate?:Date;
    ProcessOutType?:string;
    ProcessID?:number;
    PartyID?:number;
    IssueType?:string;
    WorkType?:string;
    StartDate?:Date | null;
    EndDate?:Date | null;
    DueDays:number | null = null;
    Remarks?:string;
    TotalQty?:number;
    TotalPendingQty?:number;
}

export class ProcessIn_ProcessOut_Item_ViewModel{
    WOPlanningItemComponentID: number = 0;
    WOPlanningItemID: number = 0;
    WOPlanningID: number = 0;
    WONo: string = "";
    ProcessOutItemID: number = 0;
    ProcessOutID: number = 0;
    ProcessOutNo: string = "";
    SrNo: number = 0;
    ParentProductID: number = 0;
    ProductID: number = 0;
    sku?: string;
    ProductName?: string;
    Qty: number = 0;
    UsedQty: number = 0;
    UsedDQty: number = 0;
    UsedRQty: number = 0;
    ComponentID?: number;
    ComponentName?: string;
    Remarks?: string;
    Photo?: string;
}

export class ProcessIn_ViewModel {
    ProcessInID?:number;
    ProcessInNo?:string;
    ProcessInDate?:Date;
    ProcessInType?:string;
    ProcessID?:number;
    ProcessName?: string;
    PartyID?:number;
    PartyName?: string;
    Remarks?:string;
    TotalQty?:number;
    Items:ProcessIn_Item_ViewModel[] = [];
}
export class ProcessIn_Item_ViewModel {
    ID?:number;
    WOPlanningItemComponentID: number = 0;
    ProcessOutItemID: number = 0;
    WONo: string = "";
    ProcessOutNo: string = "";
    SrNo?:number;
    ParentProductID?:number;
    ProductID?:number;
    sku?: string;
    ProductName?: string;
    ComponentID?:number;
    ComponentName?: string;
    Qty?:number;
    DQty?:number;
    RQty?:number;
    Remarks?:string;
    Photo: string = "";
}

File: HandloomAPI/Models/Sales/ProcessInModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace HandloomAPI.Models.Sales
{
    public class ProcessIn_AddModel
    {
        public long ProcessInID { get; set; }
        public string ProcessInNo { get; set; }
        public DateTime ProcessInDate { get; set; }
        public string ProcessInType { get; set; }
        public long ProcessID { get; set; }
        public long PartyID { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public List<ProcessIn_Item_AddModel> Items { get; set; }

    }
    public class ProcessIn_Item_AddModel
    {
        public long ID { get; set; }
        public long WOPlanningItemComponentID { get; set; }
        public long ProcessOutItemID { get; set; }
        public string WONo { get; set; }
        public string ProcessOutNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public long ComponentID { get; set; }
        public decimal Qty { get; set; }
        public decimal DQty { get; set; }
        public decimal RQty { get; set; }
        public string Remarks { get; set; }
    }

    public class ProcessIn_ProcessOut_ViewModel
    {
        public long ProcessOutID { get; set; }
        public string ProcessOutNo { get; set; }
        public DateTime ProcessOutDate { get; set; }
        public string ProcessOutType { get; set; }
        public long ProcessID { get; set; }
        public string ProcessName { get; set; }
        public long PartyID { get; set; }
        public string PartyName { get; set; }
        public string IssueType { get; set; }
        public string WorkType { get; set; }
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int? DueDays { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public decimal TotalPendingQty { get; set; }
    }
    public class ProcessIn_ProcessOut_Item_ViewModel
    {
        public long WOPlanningItemComponentID { get; set; }
        public long WOPlanningItemID { get; set; }
        public long WOPlanningID { get; set; }
        public string WONo { get; set; }
        public long ProcessOutItemID { get; set; }
        public long ProcessOutID { get; set; }
        public string ProcessOutNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public decimal Qty { get; set; }
        public decimal UsedQty { get; set; }
        public decimal UsedDQty { get; set; }
        public decimal UsedRQty { get; set; }
        public long ComponentID { get; set; }
        public string ComponentName { get; set; }
        public string Remarks { get; set; }
        public string Photo { get; set; }
        public List<WorkOrderPlanning_SalesOrder_Item_Attribute_ViewModel> AttributeValues { get; set; }
    }

    public class ProcessIn_ViewModel
    {
        public long ProcessInID { get; set; }
        public string ProcessInNo { get; set; }
        public DateTime ProcessInDate { get; set; }
        public string ProcessInType { get; set; }
        public long ProcessID { get; set; }
        public string ProcessName { get; set; }
        public long PartyID { get; set; }
        public string PartyName { get; set; }
        public string Remarks { get; set; }
        public decimal TotalQty { get; set; }
        public long ModifiedBy { get; set; }
        public string ModifiedByName { get; set; }
        public DateTime ModifiedDate { get; set; }
        public bool IsLocked { get; set; }
        public List<ProcessIn_Item_AddModel> Items { get; set; }

    }
    public class ProcessIn_Item_ViewModel
    {
        public long ID { get; set; }
        public long ProcessInID { get; set; }
        public long WOPlanningItemComponentID { get; set; }
        public long WOPlanningItemID { get; set; }
        public long WOPlanningID { get; set; }
        public string WONo { get; set; }
        public long ProcessOutItemID { get; set; }
        public long ProcessOutID { get; set; }
        public string ProcessOutNo { get; set; }
        public long SrNo { get; set; }
        public long ParentProductID { get; set; }
        public long ProductID { get; set; }
        public string sku { get; set; }
        public string ProductName { get; set; }
        public long ComponentID { get; set; }
        public string ComponentName { get; set; }
        public decimal Qty { get; set; }
        public decimal DQty { get; set; }
        public decimal RQty { get; set; }
        public string Remarks { get; set; }
        public string Photo { get; set; }
    }

    public class ProcessIn_Filter
    {
        public long ProcessInID { get; set; }
        public DateTime? FromDate { get; set; }
        public DateTime? ToDate { get; set; }
        public string ProcessInNo { get; set; }
    }
    
    
}
